src/compiler/scala/tools/nsc/transform/patmat/MatchTranslation.scala

```diff
@@ -262,12 +262,15 @@ trait MatchTranslation {
 
           val exSym = freshSym(pos, pureType(ThrowableTpe), "ex")
 
+          val unchecked = AnnotationInfo(UncheckedClass.tpe, Nil, Nil)
+          val scrut = atPos(pos)(Typed(REF(exSym), TypeTree(pureType(ThrowableTpe).withAnnotation(unchecked))))
```

Comment:
Alternatively I could break `combineCasesNoSubstOnly` down a bit and directly tell it that exhaustivity checking is suppressed.